### PR TITLE
test: increase timeout for stress tests

### PR DESCRIPTION
**Pull request by sdowell**
All of the stress tests are now passing in CI with the exception of TestStressLargeRequest. It turns out this test takes a much longer time to reconcile, because it is creating 13,000+ resources in a single RootSync. Increasing the default timeout gives the finalizer enough time to cleanup all of the resources.

### CODE REVIEW

**Review by haiyanmeng — APPROVED** [review]

File: e2e/testcases/stress_test.go [hunk 1]
```diff
@@ -253,8 +255,7 @@ func TestStressLargeRequest(t *testing.T) {
 		nomostest.WithRootSha1Func(nomostest.RemoteRootRepoSha1Fn),
 		nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{
 			nomostest.DefaultRootRepoNamespacedName: "configs",
-		}),
-		nomostest.WithTimeout(30*time.Minute))
```
Comment (karlkfi) [inline e2e/testcases/stress_test.go:257]: Why remove this? Doesn't this lengthen the test timeout? Don't we still need that?

File: e2e/nomostest/new.go [hunk 2]
```diff
@@ -282,8 +282,9 @@ func FreshTestEnv(t nomostesting.NTB, opts *ntopts.New) *NT {
 
 	// Longer default wait timeout for stress tests
 	// Particularly the reset/finalizer takes a long time for the stress tests
+	// Current slowest test is TestStressLargeRequest
 	if *e2e.Stress {
-		nt.DefaultWaitTimeout = nt.DefaultWaitTimeout * 2
+		nt.DefaultWaitTimeout = 30 * time.Minute
```
[the remark this answers is not in the document]
Comment (karlkfi) [reply e2e/nomostest/new.go:287]: Increasing the reconcile timeout for the test might also reduce the total time it takes, since it won't have to re-actuate every 5m.

[the remark this answers is not in the document]
Comment (sdowell) [reply e2e/nomostest/new.go:287]: The wait that needs it is in the `Reset` code that deletes the `RootSync` and [waits for `NotFound`](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/nomostest/reset.go#L203), which is handled by the test scaffolding. To configure that per-test we would need another option on `NT` to set a specific value for that timeout. I figured it wasn't worth the extra option since this is a bit of an edge case and other waits in the stress tests need the longer default timeout

File: e2e/testcases/stress_test.go (hunk 1 above)
Comment (sdowell) [reply e2e/testcases/stress_test.go:257]: I removed it since this == the default wait timeout for the stress tests now

File: e2e/nomostest/new.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (sdowell) [reply e2e/nomostest/new.go:287]: I suppose we could set a different timeout for that wait condition if `*e2e.Stress`

[the remark this answers is not in the document]
Comment (sdowell) [reply e2e/nomostest/new.go:287]: Note the stress test suite only runs a subset of ~4 targeted stress tests

[the remark this answers is not in the document]
Comment (karlkfi) [reply e2e/nomostest/new.go:287]: I guess this is fine for now. But perhaps we should refactor this in the future to add more granular timeouts so we don't get stuck in setup for 30m when something goes wrong.

[the remark this answers is not in the document]
Comment (sdowell) [reply e2e/nomostest/new.go:287]: Sorry was working on another change but was meaning to make another iteration on this. Testgrid should be green after this change but I can make a followup change to make this more granular
